tests/test_core.py

```diff
@@ -4564,13 +4564,23 @@ def test_fs_trackingdelegate(self):
     out = path_from_root('tests', 'fs', 'test_trackingdelegate.out')
     self.do_run_from_file(src, out)
 
-  @also_with_noderawfs
   def test_fs_writeFile(self, js_engines=None):
     self.emcc_args += ['-s', 'DISABLE_EXCEPTION_CATCHING=1'] # see issue 2334
     src = path_from_root('tests', 'fs', 'test_writeFile.cc')
     out = path_from_root('tests', 'fs', 'test_writeFile.out')
     self.do_run_from_file(src, out, js_engines=js_engines)
 
+  # NODERAWFS support for OSX is flaky, but it is not desirable to always
+  # disable test_fs_writeFile test, so we split the test into two.
+  # See issue 6121
+  @no_osx('NODERAWFS support on OSX is flaky')
+  @also_with_noderawfs
```

Comment:
I think that on non-mac this will run the test twice without NODERAWFS and once with it? (since `test_fs_writeFile_noderawfs` is marked as `also_with_noderawfs` which will run it once normally and once with NODERAWFS)

Seems a little wasteful, but there isn't a good and obvious way to fix it. Best I can think of is: go back to what we had before, and in the test check if `mac and is_noderawfs`, and if so skip that test (I think we need to add a method to check `is_noderawfs`).

Reply:
Not sure if I can check that, because I guess the way to implement `is_noderawfs` would be to check if `Settings.NODERAWFS == 1`. But that is true only in the child process this test_core.py launches with option `-s NODERAWFS=1` and not in test_core.py.

Reply:
Hmm, yes, I think we would need to check something like  `'NODERAWFS=1' in self.emcc_args`.

Reply:
Done